lib/tbot/tbot_test.go

```diff
@@ -43,162 +42,191 @@ func TestMain(m *testing.M) {
 	os.Exit(m.Run())
 }
 
-func rotate( //nolint:unused // used in skipped test
-	ctx context.Context, t *testing.T, log logrus.FieldLogger, svc *service.TeleportProcess, phase string,
-) {
-	t.Helper()
-	log.Infof("Triggering rotation: %s", phase)
-	err := svc.GetAuthServer().RotateCertAuthority(ctx, auth.RotateRequest{
-		// only rotate Host CA as to avoid race condition serverside when
-		// multiple CAs are rotated at once and the database closes off.
-		Type:        types.HostCA,
-		Mode:        "manual",
-		TargetPhase: phase,
-	})
-	if err != nil {
-		log.WithError(err).Infof("Error occurred during triggering rotation: %s", phase)
-	}
-	require.NoError(t, err)
-	log.Infof("Triggered rotation: %s", phase)
-}
+// TestBot is a one-shot run of the bot that communicates with a stood up
```

Comment:
This test absorbs previous tests but offers a few benefits:
- It's more external and touches less internals (e.g calling `Run` rather than invoking internal methods) - and therefore less likely to need changing as we refactor tbot.
- It tests with the fully stood up in memory auth server. 

It's quite chunky and I'd like to break it down a little bit more to make it easier to add new destinations - but I'm avoiding that now because I didn't want to change too much (*ironic I know given the size of this PR*)